src/tools/fuzzing/fuzzing.cpp

```diff
@@ -1900,9 +1906,22 @@ Expression* TranslateToFuzzReader::makeLocalSet(Type type) {
 }
 
 Expression* TranslateToFuzzReader::makeGlobalGet(Type type) {
-  auto it = globalsByType.find(type);
-  if (it == globalsByType.end() || it->second.empty()) {
-    return makeTrivial(type);
+  // In a non-function context, like in another global, we can only get from an
+  // immutable global. Whether GC is enabled also matters, as it allows getting
+  // from a non-import.
+  auto& relevantGlobals =
+    funcContext ? globalsByType
+                : (wasm.features.hasGC() ? immutableGlobalsByType
+                                         : importedImmutableGlobalsByType);
+  auto it = relevantGlobals.find(type);
+  // If we have no such relevant globals give up and emit a constant instead.
+  // We also do so if this is a tuple type and we are not in a function, because
+  // atm tuple globals have only limited support, and in particular we do not
+  // support a global.get from one such global to another (a tuple global must
+  // contain a tuple.make instruction).
```

Comment:
Hmm, that should work, but it would require us to check if we are the toplevel or not, at this location. There isn't a nice way to do that atm. Do you think it's worth it?

Reply:
I do think it's worth doing at some point, but not necessarily in this PR.

Reply:
I realized it is just as simple to disallow only `tuple.make` at the toplevel, and pushed that now. With that we can get tuple globals whose values include `global.get`s.